src/main/java/de/tum/in/www1/artemis/web/rest/UserResource.java

```diff
@@ -84,19 +87,35 @@ public UserResource(UserRepository userRepository, UserService userService, User
         this.authorityRepository = authorityRepository;
     }
 
+    private static void checkUsernameAndPasswordValidity(String username, String password) {
+
+        if (!StringUtils.hasLength(username) || username.length() < USERNAME_MIN_LENGTH) {
+            throw new AccessForbiddenException("The username has to be at least " + USERNAME_MIN_LENGTH + " characters long");
+        }
+
+        // Note: the password can be null, then a random one will be generated (Create) or it won't be changed (Update).
+        // If the password is not null, its length has to be at least PASSWORD_MIN_LENGTH
+        if (password != null && password.length() < PASSWORD_MIN_LENGTH) {
+            throw new AccessForbiddenException("The password has to be at least " + PASSWORD_MIN_LENGTH + " characters long");
+        }
+    }
+
     /**
      * POST /users : Creates a new user.
      * <p>
      * Creates a new user if the login and email are not already used, and sends an mail with an activation link. The user needs to be activated on creation.
      *
-     * @param managedUserVM the user to create
+     * @param managedUserVM the user to create. If the password is null, a random one will be generated
```

Comment:
Out of interest: Why would be generate a random password? in which cases could the password be null? As far as I see, I can neither create an account without password, nor can I send a post request to `/register` with empty password field in the request body.

Reply:
E.g. Shibboleth/SAML2 login/registration needs this, as no password is provided there.

Reply:
In the admin user interface, you can tick Random password, which set the password to `null` in the json request. I think this was developed by jhipster and not by us. We maintain it for now, but could change it in the future.

<img width="276" alt="image" src="https://user-images.githubusercontent.com/744067/122617669-f89d7680-d08c-11eb-85aa-edf0ccdc8d89.png">

Reply:
ah ok, thanks for clarification!